src/components/data/DataTable/DataTable.tsx

```diff
@@ -15,20 +15,11 @@ import TableRow from './TableRow';
 import TableRowExpandable from './TableRowExpandable';
 import DataTableContext from './DataTableContext';
 
+import type { FetchDataFunction } from '../common/types';
 import './DataTable.css';
 
-type FetchDataFunction = (params: {
-    gridParams: {
-        _search: string;
-        currentPage: number;
-        pageSize: number;
-        sortColumn: string;
-        sortAscending: boolean;
-    };
-}) => void;
-
 const defaultProps = {
-    fetchData: (() => {}) as FetchDataFunction,
+    fetchData: (() => Promise.resolve()) as FetchDataFunction,
```

Comment:
Maybe
```suggestion
const defaultProps: Partial<DataTableProps> = {
    fetchData: () => Promise.resolve(),
```
for stronger type enforcement?

Reply:
same as below: typing default props this way breaks typescript ability to make props with default value optional